src/Temporalio/Workflows/WorkflowDefinition.cs

```diff
@@ -145,6 +153,10 @@ public class WorkflowDefinition
             {
                 errs.Add("Cannot have custom name for dynamic workflow");
             }
+            if (name != null && name.Contains('\n'))
+            {
+                errs.Add("Workflow name cannot have a newline");
+            }
```

Comment:
This is a good example why the newline thing scares me.

I mean, it's dumb to have a newline in a workflow name *anyway*, but, still.

Reply:
It already occurs with metadata today, but I am preventing the newline explicitly. I don't think this use case is worth me building my own C string vec representation. But I can if I must.

Reply:
Well does that really mean we need to build our own? I've got to imagine this comes up fairly often...

And yeah, I know it's happening with metadata but the fact that we'll sometimes have to deal with edge cases like this as we add new things says to me it probably is worth a little bit of effort to just deal with that problem for good.

Reply:
I just added string arrays in [this commit](https://github.com/cretz/temporal-sdk-dotnet/commit/07524da4cfb568d61213f58420b324c5b1d10a31) and removed the newline restriction